JamfSync/Model/Jcds2Dp.swift

```diff
@@ -193,40 +201,419 @@ class Jcds2Dp: DistributionPoint {
+                    keepLooping = false
+                } else {
+                    remainingParts.append(Chunk.index)
+                    failedParts.append(Chunk.index)
+                    LogManager.shared.logMessage(message: "**** Failed to upload chunk \(Chunk.index): \(error.localizedDescription)", level: .debug)
+                }
             }
+            if uploadedChunks == Chunk.numberOf {
+                successfullUpload = true
+                keepLooping = false
+            }
+        }
+        return(successfullUpload)
+    }
+    
+    private func multipartUpload(whichChunk: Int, uploadId: String, fileUrl: URL) async -> (Result<Void, Error>) {
+        
+        LogManager.shared.logMessage(message: "Start processing part \(whichChunk)", level: .debug)
+           
+        let chunk = getChunk(fileUrl: fileUrl, part: whichChunk)
+        
+        if chunk.count == 0 {
+            return(.success(()))
+            
+        }
+        let partNumber      = whichChunk
+        let packageToUpload = fileUrl.lastPathComponent
+        
+        var urlHostAllowedPlus = CharacterSet.urlHostAllowed
+        urlHostAllowedPlus.remove(charactersIn: "+")
+        let encodedPackageName = packageToUpload.addingPercentEncoding(withAllowedCharacters: urlHostAllowedPlus) ?? ""
+        
+        let bucket          = initiateUploadData?.bucketName ?? ""
+        let region          = initiateUploadData?.region ?? ""
+        let key             = (initiateUploadData?.path ?? "") + encodedPackageName
+        let accessKeyId     = initiateUploadData?.accessKeyID ?? ""
+        let secretAccessKey = initiateUploadData?.secretAccessKey ?? ""
+        let sessionToken    = initiateUploadData?.sessionToken ?? ""
+        let jcdsServerURL   = ( region == "us-east-1" ) ? URL(string: "https://\(bucket).s3.amazonaws.com/\(key)?partNumber=\(partNumber)&uploadId=\(uploadId)")!:URL(string: "https://\(bucket).s3-\(region).amazonaws.com/\(key)?partNumber=\(partNumber)&uploadId=\(uploadId)")!
+        
+        let currentDate = Date()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyyMMdd'T'HHmmss'Z'"
+        dateFormatter.timeZone = TimeZone(identifier: "UTC")
+        let dateString = dateFormatter.string(from: currentDate)
+        
+        let urlSession: URLSession = {
+            let configuration = URLSessionConfiguration.ephemeral
+            configuration.httpShouldSetCookies = true
+            configuration.requestCachePolicy = .reloadIgnoringLocalCacheData
+            configuration.urlCache = nil
+            configuration.timeoutIntervalForRequest = 3600.0
+            configuration.timeoutIntervalForResource = 3600.0
+//            configuration.urlCache = URLCache(memoryCapacity: 0, diskCapacity: 0, diskPath: nil)
+            return URLSession(configuration: configuration)
+        }()
+        
+        var request = URLRequest(url: jcdsServerURL,
```

Comment:
This request is very similar to the one in createMultipartUpload so it would be good if they could be consolidated into one function.